dali/core/mm/perf_test.cu

```diff
@@ -0,0 +1,234 @@
+// Copyright (c) 2023, NVIDIA CORPORATION & AFFILIATES. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include <gtest/gtest.h>
+#include <chrono>
+#include <cmath>
+#include <iostream>
+#include <random>
+#include <sstream>
+#include <string>
+#include <thread>
+#include <vector>
+#include "dali/core/mm/default_resources.h"
+#include "dali/core/mm/malloc_resource.h"
+#include "dali/core/spinlock.h"
+#include "dali/core/cuda_stream.h"
+#include "dali/core/cuda_stream_pool.h"
+#include "dali/core/cuda_event.h"
+#include "dali/core/cuda_error.h"
+#include "dali/core/device_guard.h"
+
+namespace dali {
+namespace mm {
+namespace test {
+
+using perf_timer = std::chrono::high_resolution_clock;
+
+inline void print_time(std::ostream &os, double seconds) {
+  if (seconds < 1e-6) {
+    os << seconds * 1e+9 << " ns";
+  } else if (seconds < 1e-3) {
+    os << seconds * 1e+6 << " µs";
+  } else if (seconds < 1.0) {
+    os << seconds * 1e+3 << " ms";
+  } else {
+    os << seconds << " s";
+  }
+}
+
+template <typename Rep, typename Period>
+double seconds(std::chrono::duration<Rep, Period> time) {
+  return std::chrono::duration_cast<std::chrono::duration<double>>(time).count();
+}
+
+template <typename Rep, typename Period>
+void print_time(std::ostream &os, std::chrono::duration<Rep, Period> time) {
```

Comment:
`os` is unused.

Reply:
The entire function is unused. Still, it seems quite useful; I'll move it to some test utils so we don't have to reinvent it.